b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/roleItem.js

```diff
@@ -0,0 +1,131 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import CodeIcon from "@rsuite/icons/Code";
+import React, { useCallback, useEffect, useState } from "react";
+import { Nav, Panel } from "rsuite";
+import General from "./roleItemDetailsSection/general";
+import decodeGetRole from "../../../../../../util/apiDecode/settings/role/decodeGetRole";
+import AccordianItemHeader from "../../../../../common/accordianItemHeader";
+import JsonDisplay from "../../../../../common/jsonDisplay";
+
+/**
+ * 
+ * @param prop - `session`, `id`, `roleUri`
+ * 
+ * @returns role item componet
+ */
+export default function RoleItem(prop) {
+
+    const { session, id, roleUri } = prop;
+
+    const [ roleDetails, setRoleDetails ] = useState({});
+    const [ activeKeyNav, setActiveKeyNav ] = useState("1");
+
+    const fetchData = useCallback(async () => {
+        const res = await decodeGetRole(session, roleUri);
+
+        setRoleDetails(res);
+    }, [ session, roleUri ]);
+
+    useEffect(() => {
+        fetchData();
+    }, [ fetchData ]);
+
+    const activeKeyNavSelect = (eventKey) => {
+        setActiveKeyNav(eventKey);
+    };
+
+    const roleItemDetailsComponent = (activeKey) => {
+        switch (activeKey) {
+            case "1":
+
+                return <General session={ session } roleDetails={ roleDetails } fetchData={ fetchData } />;
+                // case "2":
+
+            //     return <Settings session={session} idpDetails={idpDetails} />;
+            case "4":
+
+                return <JsonDisplay jsonObject={ roleDetails } />;
+        }
+    };
+
+
+    return (
+
+        roleDetails
+            ? (<Panel
+                header={
+                    (<AccordianItemHeader
+                        title={ roleDetails.displayName }
+                        description={ `Organization role ${roleDetails.displayName} details` } />)
+                }
+                eventKey={ id }
+                id={ id }>
+                <div style={ { marginLeft: "25px", marginRight: "25px" } }>
+                    <RoleItemNav
+                        activeKeyNav={ activeKeyNav }
+                        activeKeyNavSelect={ activeKeyNavSelect } />
+                    <div>
+                        { roleItemDetailsComponent(activeKeyNav) }
+                    </div>
+                </div>
+            </Panel>)
+            : null
+    );
+}
+
+function RoleItemNav(prop) {
+
+    const { activeKeyNav, activeKeyNavSelect } = prop;
+
+    return (
+        <Nav appearance="subtle" activeKey={ activeKeyNav } style={ { marginBottom: 10, marginTop: 15 } }>
+            <div
+                style={ {
+                    alignItems: "stretch",
+                    display: "flex"
+                } }>
+                <Nav.Item
+                    eventKey="1"
+                    onSelect={ (eventKey) => activeKeyNavSelect(eventKey) }>General</Nav.Item>
+
+                { /* <Nav.Item
```

Comment:
Should we keep the commented lines? Can't we remove them if thy are not needed?

Reply:
This is already fixed when the functionality moved to a different location
https://github.com/Achintha444/samples-is/commit/537311db23374d1deff616fdcd3f1d93b7309cbe#diff-bb384cb53b82d87ca1a8aabb3f7b399758ebe30958a67a8e04fbabe56ede519a